design-proposals/persist-vmi-firmware-uuid.md

```diff
@@ -0,0 +1,91 @@
+# Overview
+This proposal introduces a mechanism to persist the firmware UUID of a Virtual Machine Instance (VMI) in KubeVirt. By storing the firmware UUID, we ensure that it remains consistent across VMI restarts, which is crucial for applications and services that rely on the UUID for identification or licensing purposes.
+
+## Motivation
+In the current implementation, the firmware UUID of a Virtual Machine Instance (VMI) is generated dynamically during the creation of the VMI. This means that if the generation logic is changed in the future, the VMI may be assigned a different firmware UUID. This behavior can cause issues for applications that depend on a stable UUID.
+Furthermore, we plan to enhance the firmware UUID generation logic to include the VM's Name and Namespace, ensuring uniqueness across different namespaces. However, changing the generation logic could inadvertently assign new UUIDs to existing VMs upon restart, leading to potential disruptions.
+By persisting the firmware UUID in the VM's status, we can maintain consistency across restarts and future-proof existing VMs against changes in the UUID generation logic.
```

Comment:
small nit: when dealing with text / markdown files, it's much easier to review if the lines are short (i.e. 1 line per sentence). The outcome markdown will look the same, but then the reviewer is able to comment with smaller granularity (that is, not select a huge amount of text for a comment).

Reply:
fixed